### PR TITLE
Add a `check_generate_docs` target

### PR DESCRIPTION
**Pull request by adrienthebo**
Currently we have no automatic way of determining if a given commit or
pull request has stale documentation; this leads to buildups in changes
that tend to tag along with other fixes.

This commit resolves the issue by adding a check that verifies that a
fresh copy of the repository does not have a diff present after docs
generation.

### CODE REVIEW

**Comment by morgante** [comment]
I'm not sure this is the best way to do this. What's wrong with just running `generate_docs` and checking if there's a diff?

**Comment by adrienthebo** [comment]
Using a clean worktree avoids issues with crossed results with the `make generate` and `make generate_docs`  tasks. If `make generate` generates a diff and then fails but documentation is current, `make generate_docs` will incorrectly fail because there's a diff present. Is there a cost to using clean worktrees that I'm not seeing?

File: test/make.sh [hunk 1]
```diff
@@ -98,3 +98,27 @@ function generate_docs() {
   done
   rm -f "$TMPFILE"
 }
+
+function check_generate() {
```
Comment (adrienthebo) [inline test/make.sh]: This should be handled in a different PR.
